### ISSUE TITLE
FAQ draft

### ISSUE THREAD

**Issue by darosior** [ElementsProject/lightning#3382]
@cdecker and I talked about introducing an FAQ, this tracks subjects/questions to add before integrating to readthedocs.

### How to get the balance of each channel ?

You can use the `listfunds` command and do a ratio of `our_amount_msat` over `amount_msat`, though this doesn't take the reserve into account.

You would be better just using the [`summary` plugin](https://github.com/lightningd/plugins/tree/master/summary) which nicely wraps these informations, along with others useful ones.

### My channel is in state `STATE`, what does that mean ?

See the [listpeers command manpage](https://lightning.readthedocs.io/lightning-listpeers.7.html#return-value).

### My payment is failing / all my payments are failing, why ?

There are a lot of possible reasons for a payment failure, but the most common one is a failure along the route from you to the payee.
The best (and also common) answer to this problem is probably "open more channels", which will increase the number of routes to the recipient, hence lowering the overall probability of a failure.
Also note to use the `pay` command which is trying all the routes for you, instead of the low-level `sendpay` command.

### How can I receive payments ?

In order to receive payments you need some incoming liquidity, which you can gain by paying through the channels you opened or by having other peers to open some channels to you.

### Can I use only one `bitcoind` for multiple `lightningd` ?

Yes, we only use `bitcoind` through the `bcli` plugin to gather Bitcoin data; we don't use `bitcoind`'s wallet, and on the same topic we don't require `txindex`.

That said, if you do that be sure to raise `bitcoind`'s max number of RPC threads, as each `lightningd` instance will use at most 4 of them (the default max).

#### Remote control

[Spark-wallet](https://github.com/shesek/spark-wallet/) is the most used remote control, and care is taken to avoid breaking it too often. **Prefer using it [behind tor](https://github.com/shesek/spark-wallet/blob/master/doc/onion.md)**.

#### `lightningd` on Android

[Some efforts](https://github.com/ElementsProject/lightning/issues/3484) are made to try to have `lightningd` running on Android, but it's nowhere near usable for now.

### How to "backup my wallet" ?

TL;DR: Backup the `hsm_secret` for your on-chain funds, on-channel funds cannot be backed up (at least for now).

As a Bitcoin user, one may be familiar with a file or a seed (or some mnemonics) from which it can recover all its funds.
C-lightning does comport a Bitcoin wallet, which you can use to make regular "on-chain" transactions, and which you can backup thanks to a seed stored in the `hsm_secret`.
In addition, C-lightning handles funds in Lightning Network channels, for which it needs to maintain a database. This database is needed for both keeping the channel operational *and to close it*, thus you can't have a single-seed backup for your funds in you channels. Backing up this database is **highly discouraged** as it can result in loss of funds because of the [penalty mechanism](https://github.com/lightningnetwork/lightning-rfc/blob/master/05-onchain.md#revoked-transaction-close-handling).

To mitigate this problem, the database replication seems to be the best approach and is currently in active development through the use of [plugin hooks](https://lightning.readthedocs.io/PLUGINS.html#db-write).

### Rescanning the block chain for lost utxos

Mentioned by cdecker in #3359    
There are 3 levels at which we can perform a rescan:     
- `rescanblockchain` in bitcoin-cli which tells bitcoind to rescan
        the blockchain starting at a given height. This does not have any
        effect on c-lightning since we track all of our information
        internally and don't rely on the wallet in 
- Starting lightningd with the --rescan=-blockheight option, which
        tells lightningd to roll back its internal state (including owned
        outputs) to the given blockheight and step through the blocks since
        then. This should detect any funds that were sent to c-lightning's
        wallet.
- Calling` dev-rescan-outputs` using lightning-cli. This will reset
        the confirmation state on outputs that have been added to the
        internal wallet, and sync it up with what bitcoind thinks about
        it. It only operates on outputs we have grabbed from the blockchain
        previously so it doesn't apply in this case where we are missing
        outputs. (*Needs `--enable-developer` configuration flag before compilation*).

So from all of these options you need to do the second, i.e., starting
    with --rescan=-blockheight with a blockheight below the height at
    which your missing output was confirmed. Don't forget to remove that
    argument later, otherwise you'll always scan from that height when
    starting :-)

### Database corruption / Channel informations lost

If you lose data (likely corrupted `lightningd.sqlite3`) about a channel __with `option_static_remotekey` enabled__, you can wait for your peer to unilateraly close the channel, then use `tools/hsmtool` with the `guesstoremote` command to (try to) recover your output from your peer's published commitment transaction.
Sadly without `option_static_remotekey`, you cant do much because keys for your output from your peer's commitment transaction are derived out of informations you lost. Fortunately, since version `0.7.3` channels are created with `option_static_remotekey` (if your peer supports it..), so roughly channels created after block [598000](https://blockstream.info/block/0000000000000000000dd93b8fb5c622b9c903bf6f921ef48e266f0ead7faedb) (short channel id starting with > 598000) have a high chance of supporting `option_static_remotekey`. You can verify it using the `listpeers` command.

**Comment by darosior** [comment]
Added some more topics I still need to fill while reading through the old issues (thanks for the labels, cdecker !!). Topics should also need to be sorted.

If we want to keep it minimal I think it's enough, did I miss an obvious F.A.Question ?
We can still fill up as we go, though.

**Comment by darosior** [comment]
I've filled up the blanks left the last time and reordered the question, I think I'll PR this soon.
